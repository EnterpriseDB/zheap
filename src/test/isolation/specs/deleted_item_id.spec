#Test case to verify concurrently deleted itemId for update

setup
{
	CREATE TABLE a (i int);
	INSERT INTO a VALUES (1), (2);
}

teardown
{
	DROP TABLE a;
}

session "s1"
setup       { BEGIN; }
step "w1"   { UPDATE a SET i = 5 WHERE i = 1; }
step "c1"   { COMMIT; }

session "s2"
setup       { BEGIN; }
step "w2"   { UPDATE a SET i = 5 WHERE i >= 1; }
step "c2"   { COMMIT; }

session "s3"
step "t3"   { DELETE FROM a WHERE i = 2; }
step "v3"   { VACUUM; }

permutation "w1" "w2" "t3" "v3" "c1" "c2"
