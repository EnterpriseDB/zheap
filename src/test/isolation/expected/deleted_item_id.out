Parsed test spec with 3 sessions

starting permutation: w1 w2 t3 v3 c1 c2
step w1: UPDATE a SET i = 5 WHERE i = 1;
step w2: UPDATE a SET i = 5 WHERE i >= 1; <waiting ...>
step t3: DELETE FROM a WHERE i = 2;
step v3: VACUUM;
step c1: COMMIT;
step w2: <... completed>
step c2: COMMIT;
